gui/wxpython/gui_core/menu.py

```diff
@@ -305,3 +308,56 @@ def OnItemSelected(self, node):
             label = data['description']
 
         self.showNotification.emit(message=label)
+
+
+class FileHistory:
+    """Add recent files history menu"""
```

Comment:
Right, as you say in the docstring, this is handling a menu, you can't use it for anything else, so add that to the name `FileHistoryMenu` or something like that (well, it is not real wx menu, but I don't have a better idea).

Reply:
> handling add file to history after save file

Works well for me! And tmp files are not saved to history because they never go through the method I guess. Great.

Reply:
FileHistory -> RecentFilesMenu